sale_discount_display_amount/hooks.py

```diff
@@ -2,35 +2,32 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 import logging
 
+from psycopg2.extensions import AsIs
+
 from odoo import SUPERUSER_ID
 from odoo.api import Environment
+from odoo.tools.sql import column_exists
 
 _logger = logging.getLogger(__name__)
 
+COLUMNS = (
+    ("sale_order", "price_total_no_discount"),
+    ("sale_order", "discount_total"),
+    ("sale_order_line", "price_total_no_discount"),
+    ("sale_order_line", "discount_total"),
+)
+
 
 def pre_init_hook(cr):
-    _logger.info("Create discount columns in database")
-    cr.execute(
-        """
-        ALTER TABLE sale_order ADD COLUMN price_total_no_discount numeric;
-    """
-    )
-    cr.execute(
-        """
-        ALTER TABLE sale_order ADD COLUMN discount_total numeric;
-    """
-    )
-    cr.execute(
-        """
-        ALTER TABLE sale_order_line ADD COLUMN price_total_no_discount
-        numeric;
-    """
-    )
-    cr.execute(
-        """
-        ALTER TABLE sale_order_line ADD COLUMN discount_total numeric;
-    """
-    )
+    for table, column in COLUMNS:
+        if not column_exists(cr, table, column):
+            _logger.info("Create discount column %s in database", column)
+            cr.execute(
+                """
+                ALTER TABLE %s ADD COLUMN %s numeric;
+            """,
+                (AsIs(table), AsIs(column)),
```

Comment:
We should use [`sql.Identifier`](https://www.psycopg.org/docs/sql.html#psycopg2.sql.Identifier) for that no?
Check this discussion on StackOverflow: https://stackoverflow.com/questions/42947885/psycopg2-difference-between-asis-and-sql-module

Reply:
hmm good point. I've never used it TBH. In this case it doesn't change anything because the values are controlled. There's nothing to escape here. Still, I'll change it because it will encourage its usage in such cases ;)

Reply:
done

Reply:
I'm using the odoo helper now, they will deal w/ future changes if any ;)